iceoryx_hoofs/source/posix_wrapper/shared_memory_object.cpp

```diff
@@ -46,68 +46,55 @@ static void memsetSigbusHandler(int) noexcept
+cxx::expected<SharedMemoryObject, SharedMemoryObjectError> SharedMemoryObjectBuilder::create() noexcept
 {
-    m_isInitialized = true;
-
-    SharedMemoryBuilder()
-        .name(name)
-        .accessMode(accessMode)
-        .openMode(openMode)
-        .filePermissions(permissions)
-        .size(m_memorySizeInBytes)
-        .create()
-        .and_then([this](auto& sharedMemory) { m_sharedMemory.emplace(std::move(sharedMemory)); })
-        .or_else([this](auto&) {
-            LogError() << "Unable to create SharedMemoryObject since we could not acquire a SharedMemory resource";
-            m_isInitialized = false;
-            m_errorValue = SharedMemoryObjectError::SHARED_MEMORY_CREATION_FAILED;
-        });
-
-    if (m_isInitialized)
+    auto printErrorDetails = [this] {
+        LogError() << "Unable to create a shared memory object with the following properties [ name = " << m_name
+                   << ", sizeInBytes = " << m_memorySizeInBytes
+                   << ", access mode = " << ACCESS_MODE_STRING[static_cast<uint64_t>(m_accessMode)]
+                   << ", open mode = " << OPEN_MODE_STRING[static_cast<uint64_t>(m_openMode)] << ", baseAddressHint = "
+                   << ((m_baseAddressHint) ? log::HexFormat(reinterpret_cast<uint64_t>(*m_baseAddressHint))
+                                           : log::HexFormat(static_cast<uint64_t>(0U)))
+                   << ((m_baseAddressHint) ? "" : " (no hint set)")
+                   << ", permissions = " << log::BinFormat(static_cast<mode_t>(m_permissions)) << " ]";
+    };
+
+    cxx::optional<SharedMemory> sharedMemory;
+    if (!SharedMemoryBuilder()
+             .name(m_name)
+             .accessMode(m_accessMode)
+             .openMode(m_openMode)
+             .filePermissions(m_permissions)
+             .size(m_memorySizeInBytes)
+             .create()
+             .and_then([&](auto& value) { sharedMemory.emplace(std::move(value)); }))
     {
-        MemoryMapBuilder()
-            .baseAddressHint((baseAddressHint) ? *baseAddressHint : nullptr)
-            .length(memorySizeInBytes)
-            .fileDescriptor(m_sharedMemory->getHandle())
-            .accessMode(accessMode)
-            .flags(MemoryMapFlags::SHARE_CHANGES)
-            .offset(0)
-            .create()
-            .and_then([this](auto& memoryMap) { m_memoryMap.emplace(std::move(memoryMap)); })
-            .or_else([this](auto) {
-                LogError() << "Failed to map created shared memory into process!";
-                m_isInitialized = false;
-                m_errorValue = SharedMemoryObjectError::MAPPING_SHARED_MEMORY_FAILED;
-            });
+        printErrorDetails();
+        LogError() << "Unable to create SharedMemoryObject since we could not acquire a SharedMemory resource";
+        return cxx::error<SharedMemoryObjectError>(SharedMemoryObjectError::SHARED_MEMORY_CREATION_FAILED);
     }
```

Comment:
In our examples and tests we often use this pattern
```cpp
auto sharedMemoryResult = SharedMemoryBuilder()
             .name(m_name)
             // ...
             .create();
if (sharedMemoryResult.has_error())
{
    return ...;
}
auto& sharedMemory = sharedMemoryResult.value();
```
This is as readable as the existing code but does not copy/move and does not use additional stack memory.